sds/src/secondary_validation/sweden_pin_checksum.rs

```diff
@@ -0,0 +1,72 @@
+pub struct SwedenPINChecksum;
+use crate::secondary_validation::Validator;
+
+const MULTIPLIERS: [u32; 9] = [2, 1, 2, 1, 2, 1, 2, 1, 2];
+
+fn sum_of_digits(digits: u32) -> u32 {
+    let mut sum = 0;
+    let mut num = digits;
+    while num > 0 {
+        sum += num % 10;
+        num /= 10;
+    }
+    sum
+}
+
+impl Validator for SwedenPINChecksum {
+    fn is_valid_match(&self, regex_match: &str) -> bool {
+        /*
+         * Swedish Persional Identification Number uses a 13-digit identification code.
+         * The first 9 digits are multiplied by a multiplier from the MULTIPLIERS array.
+         * Each product has its digits summed up and and added to a total.
+         * The total is used to compute a checksum, which is compared to the last digit of the input.
+         */
+
+        let mut total = 0;
+        let mut index = 0;
+        let mut checksum_digit = 0;
+
```

Comment:
If we even add a optional separators in the regex, then the checksum will fail. To make the checksum a bit more flexible, any non valid character can be filtered out with         `let valid_chars = regex_match.chars().filter(|c| c.is_ascii_digit())`;

Reply:
I decided to use `filter_map` to simplify it even more